tests/unit/fem/test_lin_interp.cpp

```diff
@@ -59,164 +96,1226 @@ double FdotG3(const Vector & x)
+   int order_h1 = 1, order_l2 = 1, n = 3, dim = 1;
    double tol = 1e-9;
 
-   Mesh mesh(n, n, n, Element::HEXAHEDRON, 1, 2.0, 3.0, 5.0);
+   FunctionCoefficient     fCoef(f1);
 
-   FunctionCoefficient       fCoef(f3);
+   for (int type = (int)Element::SEGMENT;
+        type <= (int)Element::SEGMENT; type++)
+   {
+      Mesh mesh(n, (Element::Type)type, 1, 2.0);
+
+      SECTION("Operators on H1 for element type " + std::to_string(type))
+      {
+         H1_FECollection    fec_h1(order_h1, dim);
+         FiniteElementSpace fespace_h1(&mesh, &fec_h1);
+
+         GridFunction f0(&fespace_h1);
+         f0.ProjectCoefficient(fCoef);
+
+         SECTION("Mapping to L2")
+         {
+            L2_FECollection    fec_l2(order_l2, dim);
+            FiniteElementSpace fespace_l2(&mesh, &fec_l2);
+
+            GridFunction f1(&fespace_l2);
+
+            DiscreteLinearOperator Op(&fespace_h1,&fespace_l2);
+            Op.AddDomainInterpolator(new IdentityInterpolator());
+            Op.Assemble();
+
+            Op.Mult(f0,f1);
+
+            REQUIRE( f1.ComputeL2Error(fCoef) < tol );
+         }
+         SECTION("Mapping to L2 (INTEGRAL)")
+         {
+            L2_FECollection    fec_l2(order_l2, dim,
+                                      BasisType::GaussLegendre,
+                                      FiniteElement::INTEGRAL);
+            FiniteElementSpace fespace_l2(&mesh, &fec_l2);
+
+            GridFunction f1(&fespace_l2);
+
+            DiscreteLinearOperator Op(&fespace_h1,&fespace_l2);
+            Op.AddDomainInterpolator(new IdentityInterpolator());
+            Op.Assemble();
+
+            Op.Mult(f0,f1);
+
+            REQUIRE( f1.ComputeL2Error(fCoef) < tol );
+         }
+      }
+   }
+}
+
+TEST_CASE("2D Identity Linear Interpolators",
+          "[IdentityInterpolator]")
+{
+   int order_h1 = 1, order_nd = 2, order_rt = 1, order_l2 = 1, n = 3, dim = 2;
+   double tol = 1e-9;
+
+   FunctionCoefficient     fCoef(f2);
+   VectorFunctionCoefficient FCoef(dim, F2);
+
+   for (int type = (int)Element::TRIANGLE;
+        type <= (int)Element::QUADRILATERAL; type++)
+   {
+      Mesh mesh(n, n, (Element::Type)type, 1, 2.0, 3.0);
+
+      SECTION("Operators on H1 for element type " + std::to_string(type))
+      {
+         H1_FECollection    fec_h1(order_h1, dim);
+         FiniteElementSpace fespace_h1(&mesh, &fec_h1);
+
+         GridFunction f0(&fespace_h1);
+         f0.ProjectCoefficient(fCoef);
+
+         SECTION("Mapping to H1")
+         {
+            H1_FECollection    fec_h1p(order_h1+1, dim);
+            FiniteElementSpace fespace_h1p(&mesh, &fec_h1p);
+
+            GridFunction f0p(&fespace_h1p);
+
+            DiscreteLinearOperator Op(&fespace_h1,&fespace_h1p);
+            Op.AddDomainInterpolator(new IdentityInterpolator());
+            Op.Assemble();
+
+            Op.Mult(f0,f0p);
+
+            REQUIRE( f0p.ComputeL2Error(fCoef) < tol );
```

Comment:
Here and in some other cases, it would be more appropriate to check the H1/Hcurl/Hdiv error instead of the L2.

Reply:
OK, this sounds good.  Should we uses these measures whenever the target spaces allow them or should we be more selective?

Reply:
I am thinking using the norm induced by the range space. This will also require a small change in 2D Hcurl problems where the CurlCoeff will have to be changed from Scalar to Vector of dim = 1.

Reply:
I think I found them all.  This was a good idea.  Thanks, @psocratis !